massa-execution-worker/src/execution.rs

```diff
@@ -385,13 +420,27 @@ impl ExecutionState {
         &self,
         operation: &OperationType,
         buyer_addr: Address,
+        current_slot: Slot,
     ) -> Result<(), ExecutionError> {
         // process roll buy operations only
         let roll_count = match operation {
             OperationType::RollBuy { roll_count } => roll_count,
             _ => panic!("unexpected operation type"),
         };
 
+        // control vesting max_rolls for buyer address
+        if let Some(vesting_range) = self.find_vesting_range(&buyer_addr, &current_slot) {
+            let rolls = self.get_final_and_candidate_rolls(&buyer_addr);
+            // (candidate_rolls + amount to buy)
+            let max_rolls = rolls.1.saturating_add(*roll_count);
+            if max_rolls > vesting_range.max_rolls {
+                return Err(ExecutionError::VestingError(format!(
+                    "vesting_max_rolls={} with value max_rolls={} ",
```

Comment:
those messages would gain from being clearer:

"trying to get to a total of XXX rolls but only YYY are allowed at that time by the vesting scheme"